File: sdk/analiseRisco.py
// analiseRisco.js

function analisarRisco(dados) {
  let score = 0;

  if (dados.tempo_na_pagina < 3) score += 30;
  if (!dados.movimento_mouse) score += 30;
  if (dados.dispositivo_novo) score += 20;
  if (dados.ip_suspeito) score += 20;

  let action = 'allow';
  if (score > 75) action = 'deny';
  else if (score > 30) action = 'review';

  return { score, action };
}

module.exports = analisarRisco;

File: sdk/analiseRisco.js
const regras = require('./regras');
const registrarEvento = require('./logger'); // ✅ IMPORTADO

function analisarRisco(dados) {
  let score = 0;

  regras.forEach(regra => {
    if (regra.condicao(dados)) {
      score += regra.peso;
    }
  });

  const action = score > 75 ? 'deny' : score > 30 ? 'review' : 'allow';

  const resultado = { score, action };
  registrarEvento({ ...dados, ...resultado }); // ✅ REGISTRANDO

  return resultado;
}

module.exports = analisarRisco;
